pkg/controllers/aws_infra.go

```diff
@@ -85,12 +84,10 @@ func (r *HypershiftDeploymentReconciler) createAWSInfra(hyd *hypdeployment.Hyper
 		}
 		log.Info("Infrastructure configured")
 
-		if err := r.Get(ctx, types.NamespacedName{Namespace: hyd.Namespace, Name: hyd.Name}, hyd); err != nil {
```

Comment:
This is not needed, as the patchHypershiftDeploymentResource above, does a Client.Get and then patches it, so the returned HypershiftDeployment will be the latest version of the resource.